### PR TITLE
fix: added support for label text wrap in cartesian chart

### PR DESCRIPTION
**Pull request by Greedy-Geek**
## Description
Added support for label text wrap in cartesian chart. 
Updated the`wrapTextIfNeeded()` in `svg-util.service.ts` to wrap text correctly.

### Checklist:
- My changes generate no new warnings

### CODE REVIEW

File: projects/observability/src/shared/components/utils/svg/svg-util.service.ts [hunk 1]
```diff
@@ -150,13 +150,12 @@ export class SvgUtilService {
...
-
+          const y = textSelection.attr("y");
+          const dy = parseFloat(textSelection.attr("dy"));
           currentLine.pop();
           tspan.text(currentLine.join(' '));
           currentLine = [word];
-          tspan = appendTSpan()
-            .attr('dy', `${++lineNumber * 1.1}em`)
-            .text(word);
+          tspan = textSelection.append("tspan").attr("x", 0).attr("y", y).attr("dy", `${++lineNumber * 1.1 + dy}em`).text(word)
```
Comment (anandtiwary) [inline projects/observability/src/shared/components/utils/svg/svg-util.service.ts:158]: how is `appendTSpan` different than `textSelection.append("tspan")` ?

Comment (Greedy-Geek) [reply projects/observability/src/shared/components/utils/svg/svg-util.service.ts:158]: Oh sorry my mistake there 😅. No difference there, the change is after that, i.e., for the gap before the text in below line, and its line-height. Updating

Comment (Greedy-Geek) [reply projects/observability/src/shared/components/utils/svg/svg-util.service.ts]: updated

File: projects/observability/src/shared/components/utils/svg/svg-util.service.ts [hunk 2]
```diff
@@ -150,13 +150,12 @@ export class SvgUtilService {
...
-
+          const y = textSelection.attr("y");
+          const dy = parseFloat(textSelection.attr("dy"));
           currentLine.pop();
           tspan.text(currentLine.join(' '));
           currentLine = [word];
-          tspan = appendTSpan()
-            .attr('dy', `${++lineNumber * 1.1}em`)
-            .text(word);
+          tspan = appendTSpan().attr("x", 0).attr("y", y).attr("dy", `${++lineNumber * 1.1 + dy}em`).text(word)
```
Comment (anandtiwary) [inline projects/observability/src/shared/components/utils/svg/svg-util.service.ts:158]: nit: put `.<method>` on a new line

**Review by anandtiwary — APPROVED** [review]

File: projects/observability/src/shared/components/utils/svg/svg-util.service.ts (hunk 2 above)
Comment (Greedy-Geek) [reply projects/observability/src/shared/components/utils/svg/svg-util.service.ts]: oh, I thought if prettier doesn't complain I shall leave it that way. Updated

**Review by anandtiwary — APPROVED** [review]

**Review by adisreyaj — APPROVED** [review]

**Pull request merged**